lib/govuk_publishing_components/presenters/public_layout_helper.rb

```diff
@@ -46,17 +46,48 @@ class PublicLayoutHelper
       attr_reader :footer_navigation, :footer_meta, :cookie_banner_data
 
       def initialize(local_assigns)
+        @footer_links_count = generate_ga4_footer_links_count(local_assigns)
         @footer_navigation = local_assigns[:footer_navigation] || navigation_link_generation_from_locale(I18n.t("components.layout_footer.navigation_links"))
-        @footer_meta = local_assigns[:footer_meta] || { items: add_data_attributes_to_links(FOOTER_META[:items], "supportLink") }
+        @footer_meta = local_assigns[:footer_meta] || { items: add_data_attributes_to_links(FOOTER_META[:items], "supportLink", { section: "Support links", index_section: 3 }) }
         @cookie_banner_data = local_assigns[:cookie_banner_data] || {}
       end
 
+      def generate_ga4_footer_links_count(local_assigns)
+        footer_links_count = 0
+
+        footer_nav = local_assigns[:footer_navigation] || I18n.t("components.layout_footer.navigation_links")
+
+        footer_nav.each do |column|
+          if column[:items]
+            footer_links_count += column[:items].length
+          elsif column[:menu_contents]
+            footer_links_count += column[:menu_contents].length
+          end
+        end
+
+        footer_meta = local_assigns[:footer_meta] || FOOTER_META
+        footer_links_count += footer_meta[:items].length
+
+        extra_footer_links = 2 # Amount of links to add to @footer_links_count. This accounts for the OGL link and Crown Copyright link.
```

Comment:
Not sure you need to declare this as a variable as it only gets used on the next line?

```Ruby
# add 2 to the links count to account for the OGL link and Crown Copyright link.
footer_links_count += 2
```

Reply:
@andysellick Cool, wasn't sure which method to use for this but I will do that one instead :+1:

Reply:
Good stuff. Also I've checked and you don't need to explicitly return the variable by itself as the last line in the method, whatever the last line equates to is what is returned, i.e.

```Ruby
def test
  thing = 5
  thing + 5
  thing
end
```

is equivalent to

```Ruby
def test
  thing = 5
  thing + 5
end
```

(they both return `10`)